lamindb/models/schema.py

```diff
@@ -578,6 +578,13 @@ def __init__(
                 self.optionals.set(optional_features)
                 return None
         self._slots: dict[str, Schema] = {}
+        # if both features and a schema are provided, we use a slot for a new schema of the features
+        if features and slots:
+            main_schema = Schema(features=features).save()
```

Comment:
This is not OK because you're saving an object to the database in the constructor. We're avoiding this everywhere else in the codebase. When the user calls `Schema()`, they expect this to be idempotent aka not mutate the database. This has to be changed.

However, I don't understand the rationale for the design choice of implicitly creating a schema component in the first place.

I think I just don't get the design. 🤔 I'm tempted to argue along the parallel argument that a dataframe schema should, IMO, respect the hierarchy of the data structure and not introduce artificial hierarchy. But I'm not sure whether this applies.

I'll try to understand this better by looking at examples.

Reply:
So, it fires in this example:

<img width="630" height="75" alt="image" src="https://github.com/user-attachments/assets/c5d2c88d-5774-49d0-b090-ee604deb4d97" />

And I'm not pretty sure that the feedback I gave for https://github.com/laminlabs/lamindb/pull/3036 applies here, too: we should keep the number of schema records minimal to keep the database complexity low. So, I'm pretty sure that we should _not_ create an additional schema (and definitely not implicitly save it to the database).

Why did you choose the design you chose?